extensions/resource-deployment/src/ui/notebookWizard/notebookWizard.ts

```diff
@@ -49,15 +48,32 @@ export class NotebookWizard extends WizardBase<NotebookWizard, NotebookWizardPag
 
 	protected initialize(): void {
 		this.setPages(this.getPages());
-		this.wizardObject.generateScriptButton.hidden = true;
-		this.wizardInfo.actionText = this.wizardInfo.actionText || localize('notebookWizard.ScriptToNotebook', "Script to Notebook");
-		this.wizardObject.doneButton.label = this.wizardInfo.actionText;
 	}
 
 	protected onCancel(): void {
 	}
 
+	protected async onGenerateScript(): Promise<void> {
+		const { notebook, env } = await this.prepareNotebookAndEnvironment();
+		try {
+			Object.assign(process.env, env);
```

Comment:
Can you check whether this is going to be an issue for scripting multiple deployment notebooks? Specifically this scenario :

1. Script MIAA deployment A to Notebook, but don't execute it
2. Script MIAA deployment B to Notebook with different env vars (password is an easy one), but don't execute it
3. Go back and execute MIAA deployment Notebook A

Will A still be using its own values or will it have been overwritten with B's?

If it is overwritten we could solve that by appending an identifier onto the env var names - just do a counter that starts at 1 and gets incremented every time a Notebook is deployed (and do a search/replace in the Notebook prior to scripting)

If you do have to do that let's do it as a separate PR so as to not clutter up this one.

Reply:
Like your suggestion. I will do this test. However, GenerateScript is the default action in main currently so this issue/test can be addressed independently of this PR.

Reply:
This test is complete. The passwords work as expected, so we do not need to make any additional changes.